pkg/controller/import-controller.go

```diff
@@ -44,315 +53,536 @@ const (
-		cdiClient:  cdiClientSet,
-		Controller: *NewController(client, pvcInformer, podInformer, image, pullPolicy, verbose),
-		recorder:   recorder,
-	}
-	return c
-}
-
-func (ic *ImportController) findScratchPvcFromCache(pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
-	return ic.pvcLister.PersistentVolumeClaims(pvc.Namespace).Get(pvc.Name + "-scratch")
-}
-
-func (ic *ImportController) findImportPodFromCache(pvc *v1.PersistentVolumeClaim) (*v1.Pod, error) {
-	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: map[string]string{LabelImportPvc: pvc.Name}})
+// NewImportController creates a new instance of the import controller.
+func NewImportController(mgr manager.Manager, cdiClient *cdiclientset.Clientset, k8sClient kubernetes.Interface, log logr.Logger, importerImage, pullPolicy, verbose string) (controller.Controller, error) {
+	reconciler := &ImportReconciler{
+		Client:     mgr.GetClient(),
+		Scheme:     mgr.GetScheme(),
+		CdiClient:  cdiClient,
+		K8sClient:  k8sClient,
+		Log:        log.WithName("import-controller"),
+		Image:      importerImage,
+		Verbose:    verbose,
+		PullPolicy: pullPolicy,
+		recorder:   mgr.GetRecorder("import-controller"),
+	}
+	importController, err := controller.New("import-controller", mgr, controller.Options{
+		Reconciler: reconciler,
+	})
 	if err != nil {
 		return nil, err
 	}
-
-	podList, err := ic.podLister.Pods(pvc.Namespace).List(selector)
-	if err != nil {
+	if err := addImportControllerWatches(mgr, importController); err != nil {
 		return nil, err
 	}
+	return importController, nil
+}
 
-	if len(podList) == 0 {
-		return nil, nil
-	} else if len(podList) > 1 {
-		return nil, errors.Errorf("multiple pods found for import PVC %s/%s", pvc.Namespace, pvc.Name)
+func addImportControllerWatches(mgr manager.Manager, importController controller.Controller) error {
+	// Add schemes.
+	if err := corev1.AddToScheme(mgr.GetScheme()); err != nil {
+		return err
 	}
-	return podList[0], nil
-}
 
-func (ic *ImportController) requiresScratchSpace(pvc *v1.PersistentVolumeClaim) bool {
-	scratchRequired := false
-	contentType := getContentType(pvc)
-	// All archive requires scratch space.
-	if contentType == "archive" {
-		scratchRequired = true
-	} else {
-		switch getSource(pvc) {
-		case SourceGlance:
-			scratchRequired = true
-		case SourceRegistry:
-			scratchRequired = true
-		}
+	// Setup watches
+	if err := importController.Watch(&source.Kind{Type: &corev1.PersistentVolumeClaim{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
 	}
-	value, ok := pvc.Annotations[AnnRequiresScratch]
-	if ok {
-		boolVal, _ := strconv.ParseBool(value)
-		scratchRequired = scratchRequired || boolVal
+	if err := importController.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		OwnerType:    &corev1.PersistentVolumeClaim{},
+		IsController: true,
+	}, predicate.Funcs{
+		DeleteFunc: func(e event.DeleteEvent) bool {
```

Comment:
Shouldn't we be handling deletes?  If someone deletes an import pod, we want to recreate it, right?

Reply:
Good point, I think I added that to stop the loop that would keep creating a new importer pod once it finished, because we would delete the completed pod. It should not be needed anymore now that I have solved that.

Reply:
Yeah don't need this, removed.